RubTextEditor/TextEditor: Make ctrl+backspace delete the previous word

fmqa opened the pull request:
Fixes #2914 

* Ctrl+Backspace now deletes the previous word
* Ctrl+Delete now deletes the next word (note that this never worked, even with the previous key-binding shift+delete due to an off-by-one error)
* Updates PharoShortcuts accordingly (I am aware that they are currently unused - this is noted by #11191, but it is important to keep these updated as we may want to use them at some point)

fmqa committed:
RubTextEditor: Make ctrl+backspace delete word

fmqa committed:
TextEditor: Make ctrl+backspace delete word

fmqa committed:
PharoShortcuts: Update word editing shortcuts

These are currently unused, but update them for later usage

fmqa committed:
RubTextEditor: Make ctrl+delete delete the next word

This never worked even with the previous (shift + delete) keybinding due to an off-by-one error

fmqa committed:
TextEditor: Make ctrl+delete delete the next word

Ducasse commented:
Thanks we will have a look. This is great to see some help on this side.

VincentBlondeau commented on src/Tool-Base/PharoShortcuts.class.st:
```diff
@@ -249,7 +249,7 @@ PharoShortcuts >> forwardLineShortcut [
 { #category : #'keymaps - editor navigation' }
 PharoShortcuts >> forwardWordShortcut [
-	^ Character arrowRight meta
+	^ Character arrowRight ctrl
```
I'm not sure that this change will work on Mac.

fmqa replied:
Changed + Added comment

VincentBlondeau commented on src/Tool-Base/PharoShortcuts.class.st:
```diff
@@ -462,7 +462,7 @@ PharoShortcuts >> selectBackwardLineShortcut [
 { #category : #'keymaps - editor selection' }
 PharoShortcuts >> selectBackwardWordShortcut [
-	^ Character arrowLeft meta shift
+	^ Character arrowLeft shift ctrl
```
I'm not sure that this change will work on Mac.

fmqa replied:
Changed + Added comment

VincentBlondeau commented on src/Tool-Base/PharoShortcuts.class.st:
```diff
@@ -479,7 +479,7 @@ PharoShortcuts >> selectForwardLineShortcut [
 { #category : #'keymaps - editor selection' }
 PharoShortcuts >> selectForwardWordShortcut [
-	^ Character arrowRight shift meta
+	^ Character arrowRight shift ctrl
```
I'm not sure that this change will work on Mac.

fmqa replied:
Changed + Added comment

VincentBlondeau commented on src/Tool-Base/PharoShortcuts.class.st:
```diff
@@ -174,7 +174,7 @@ PharoShortcuts >> delShortcut [
 PharoShortcuts >> delWordShortcut [
 
-	^ Character delete shift asShortcut
+	^ Character delete ctrl asShortcut
```
Good for Windows. Not sure about Mac support.

fmqa replied:
> Control-D: Delete the character to the right of the insertion point. Or use Fn-Delete.
> Fn-Delete: Forward delete on keyboards that don't have a Forward Delete   key. Or use Control-D.

Not sure how to map this

fmqa replied:
We can make it conditional for now.

VincentBlondeau commented on src/Tool-Base/PharoShortcuts.class.st:
```diff
@@ -55,7 +55,7 @@ PharoShortcuts >> backwardLineShortcut [
 PharoShortcuts >> backwardWordDeleteShortcut [
 
-	^ Character backspace shift
+	^ Character backspace ctrl
```
Good for Windows. Not sure about Mac support.

fmqa replied:
Changed + Added comment

fmqa committed:
PharoShortcuts >> forwardWordShortcut: Change it back to meta+right

meta+right on Win/Linux: ctrl+right
meta+right on OSX: option+right

also add a comment

fmqa committed:
PharoShortcuts >> selectBackwardWordShortcut: Use meta instead of ctrl for Mac compatibility

fmqa committed:
PharoShortcuts >> selectForwardWordShortcut: Use meta key for Mac compatibility

fmqa committed:
PharoShortcuts >> backwardWordDeleteShortcut: Use meta for Mac compatibility

fmqa committed:
PharoShortcuts >> delWordShortcut: Add conditional logic

fmqa committed:
PharoShortcuts: Add missing parens in delWordShortcut

fmqa committed:
PharoShortcuts >> delWordShortcut: Fix typo

NSUSpray commented on src/Tool-Base/PharoShortcuts.class.st:
```diff
@@ -249,6 +251,7 @@ PharoShortcuts >> forwardLineShortcut [
 
 { #category : #'keymaps - editor navigation' }
 PharoShortcuts >> forwardWordShortcut [
+	"Equivalent to: ctrl+right on Windows/Linux, option/alt+right on OSX"
```
It seems you forgot to add the same comment for the _**backward**WordShortcut_ method.

fmqa replied:
Thanks for catching this. On my phone now (holidays...) but this can be fixed quickly.

NSUSpray commented:
It works fine for me.

fmqa committed:
PharoShortcuts: Add missing comment to backwardWordShortcut

fmqa commented:
I think this is about ready at this point. Would be great if Dr. @VincentBlondeau or an OSX user signs off on it.

Ducasse commented:
Broken tests are unrelated.
